syntax/grammar.mjs

```diff
@@ -373,48 +373,59 @@ let Function =
 /* ---------- */
 /* Characters */
 
-let backslash = string("\\");
-let quote = string("\"");
-
-/* Any Unicode character from BMP excluding C0 control characters, space,
+/* Any Unicode character excluding C0 control characters (but including tab),
  * surrogate blocks and non-characters (U+FFFE, U+FFFF).
- * Cf. https://www.w3.org/TR/REC-xml/#NT-Char
- */
-let regular_char =
-    charset("\\u{21}-\\u{D7FF}\\u{E000}-\\u{FFFD}\\u{10000}-\\u{10FFFF}");
+ * Cf. https://www.w3.org/TR/REC-xml/#NT-Char */
+let any_char =
+    either(
+        charset("\\u{9}\\u{20}-\\u{D7FF}\\u{E000}-\\u{FFFD}"),
+        charset("\\u{10000}-\\u{10FFFF}"));
 
-let text_char = defer(() =>
+/* The opening brace in text starts a placeable. */
+let special_text_char =
+    string("{");
+
+/* Double quote and backslash need to be escaped in string literals. */
+let special_quoted_char =
     either(
-        blank_inline,
-        string("\u0009"),
-        regex(/\\u[0-9a-fA-F]{4}/),
-        sequence(
-            backslash,
-            backslash).map(join),
-        sequence(
-            backslash,
-            string("{")).map(join),
-        and(
-            not(backslash),
-            not(string("{")),
-            regular_char)));
+        string("\""),
+        string("\\"));
+
+let text_char =
+    and(
+        not(special_text_char),
+        any_char);
 
-let indented_char = defer(() =>
+/* Indented text may not start with characters which mark its end. */
+let indented_char =
     and(
         not(string(".")),
         not(string("*")),
         not(string("[")),
         not(string("}")),
-        text_char));
+        text_char);
```

Comment:
This needs to be `text_char` so that the parser doesn't consume the `{` of placeables on a new line as `indented_char`.

```properties
message =
    Foo Bar
    {placeable}.
```

Reply:
My intent was

```
  and(
        not(string(".")),
        not(string("*")),
        not(string("[")),
        not(string("{")),
        not(string("}")),
        any_char);
```
thus creating symmetry.

Reply:
I prefer to use `text_char` here. `indented_char` is a specialized version of it, after all. Plus, #186 might change this code anyways.